rpc/server_auth.go

```diff
@@ -277,70 +274,48 @@ func (ss *simpleServer) ensureAuthed(ctx context.Context) (string, interface{},
 		}
 
 		return &ss.authRSAPrivKey.PublicKey, nil
-	})
-	if err != nil {
-		return "", nil, status.Errorf(codes.Unauthenticated, "unauthenticated: %s", err)
+	}); err != nil {
+		return nil, status.Errorf(codes.Unauthenticated, "unauthenticated: %s", err)
 	}
 
-	// By default use the standard rpc.JWTClaims
-	var claims Claims = &JWTClaims{}
-
-	// If AuthHandler is using CustomClaims use the claims type provided.
-	if provider, ok := handler.(TokenCustomClaimProvider); ok {
-		// reset the claims to the handlers version
-		claims = provider.CreateClaims()
-		if claims == nil {
-			return "", nil, status.Error(codes.Internal, "invalid implementation of TokenCustomClaimProvider, cannot return nil")
+	// Audience verification is critical for security. Without it, we have a higher chance
+	// of validating a JWT is valid, but not that it is intended for us. Of course, that means
+	// we trust whomever owns the private keys to signing access tokens.
+	audVerified := false
+	for _, allowdAud := range ss.authAudience {
+		if claims.RegisteredClaims.VerifyAudience(allowdAud, true) {
```

Comment:
This is new and will require some robot/client restarts. I don't think it will cause any issues but flagging it.

Reply:
[minor] To reduce issues we could just `VerifyAudience(allowdAud) || VerifyAudience(claims.RegisteredClaims.Subject)` for 24hrs and then remove the 2nd check in a cleanup PR?

Reply:
I'm not sure that would fix it. We are now in general verifying all audiences so I think your suggestion would still fail?

Reply:
Ok. I was just looking at the diff. previously the Audience attribute was set to `Audience:  jwt.ClaimStrings{entity}` and now is set to `Audience:  audience` on https://github.com/viamrobotics/goutils/pull/145/files#diff-988c4b375bc405125e428ecc73b8b3ee9da068e30a541214b93d20fced98e856R139

So I thought the incompatibility you were talking about is that for 24hrs the wrong audience value would be validated.